PackageManagement/nuget.yml

```diff
@@ -0,0 +1,111 @@
+name: Push Nuget Package to Azure Artifacts or GitHub Package Registry
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - '*'
+
+# CONFIGURATION
+#
+# 1. Set the environment variables for your Build Configuration (Release/Debug), .NET Version,
+#    GitHub Package Registry URL and Azure Artifacts Feed URL as appropriate.
+#
+# 2. Ensure that your csproj file is configured correctly as outlined in the documentation below:
+#    https://docs.github.com/en/free-pro-team@latest/packages/using-github-packages-with-your-projects-ecosystem/configuring-dotnet-cli-for-use-with-github-packages#publishing-a-package
+#
+# 3. Set up the following secrets in your repository:
+#    AZURE_ARTIFACTS_PAT
+#    
+#    You can follow the below documentation to create a Personal Access Token (PAT)
+#    https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page#create-a-pat
+#
+#    You can find additional information on the scopes and associated permissions in the documentation below:
+#    https://docs.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/oauth?view=azure-devops#scopes
+#
+
+env:
+  AZURE_ARTIFACTS_FEED_URL: https://pkgs.dev.azure.com/.../nuget/v3/index.json            # set this to the registry url for your Azure Artifacts Feed, you will find this in Azure DevOps
+  BUILD_CONFIGURATION: 'Release'                                                          # set this to the appropriate build configuration
+  DOTNET_VERSION: '3.1.x'                                                                 # set this to the dot net version to use
+  GITHUB_PACKAGE_REGISTRY_URL: https://nuget.pkg.github.com/yourorganisation/index.json   # set this to the registry url for your GitHub package registry
```

Comment:
Is this an optional parameter? Is there a possibility to set a default value for this by using $GITHUB_REPOSITORY variable? or if user doesnt configure this, would registry URL automatically point to the current organization?

Reply:
Per the sample at https://github.com/actions/setup-dotnet, this is something that will need to be set. However, this is at the user/organisation level, and not on a per repository basis.

We *could* change to GITHUB_ACTOR, but i think that would actually restrict this workflow. e.g. if I made a change which triggered a workflow in Azure/actions-workflow-samples, then GITHUB_ACTOR would be christianreddington and not Azure.

To keep it flexible and work in most cases, i think it's best to keep as it is. However, it could be good product feedback for GitHub to provide more a variable for just the organization/account that you're deploying into?

Reply:
Sure, I'll pass the feedback. And for the current scenario, I agree that <yourorganisation> can be a placeholder.